server/plugin.go

```diff
@@ -62,73 +67,95 @@ func (p *Plugin) OnActivate() error {
 		return errors.Wrap(err, "failed to ensure bot account")
 	}
 
+	mmconf := p.mm.Configuration.GetConfig()
+	p.conf = configurator.NewConfigurator(p.mm, p.BuildConfig, botUserID)
 	stored := api.StoredConfig{}
 	_ = p.mm.Configuration.LoadPluginConfiguration(&stored)
+	err = p.conf.Reconfigure(&stored)
+	if err != nil {
+		return errors.Wrap(err, "failed to reconfigure configurator on startup")
+	}
+	conf := p.conf.GetConfig()
 
 	accessKey := os.Getenv("APPS_INVOKE_AWS_ACCESS_KEY")
 	if accessKey == "" {
-		mm.Log.Warn("APPS_INVOKE_AWS_ACCESS_KEY is not set. AWS apps won't work.")
+		p.mm.Log.Warn("APPS_INVOKE_AWS_ACCESS_KEY is not set. AWS apps won't work.")
 	}
 	secretKey := os.Getenv("APPS_INVOKE_AWS_SECRET_KEY")
 	if secretKey == "" {
-		mm.Log.Warn("APPS_INVOKE_AWS_SECRET_KEY is not set. AWS apps won't work.")
+		p.mm.Log.Warn("APPS_INVOKE_AWS_SECRET_KEY is not set. AWS apps won't work.")
 	}
-
-	awsClient := aws.NewAWSClient(accessKey, secretKey, &mm.Log)
-
-	conf := configurator.NewConfigurator(mm, awsClient, p.BuildConfig, botUserID)
-	err = conf.RefreshConfig(&stored)
+	p.aws, err = aws.MakeService(accessKey, secretKey, &p.mm.Log)
 	if err != nil {
-		return errors.Wrap(err, "failed to refresh config on startup")
+		return errors.Wrap(err, "failed to initialize AWS access")
 	}
-	store := store.New(mm, conf)
-	proxy := proxy.NewProxy(mm, awsClient, conf, store)
 
-	mutex, err := cluster.NewMutex(p.API, mutexKey)
+	p.store = store.New(p.mm, p.conf)
+	// manifest store
+	mstore := p.store.Manifest()
+	err = mstore.Configure(conf)
 	if err != nil {
-		return errors.Wrapf(err, "failed creating cluster mutex")
+		return errors.Wrap(err, "failed to initialize the manifest store")
+	}
+	manifestBucket := awsclient.GenerateS3BucketNameWithDefaults("")
+	err = mstore.InitGlobal(p.aws.Client(), manifestBucket)
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize the global manifest list from marketplace")
+	}
+	// app store
+	appstore := p.store.App()
+	if pluginapi.IsConfiguredForDevelopment(mmconf) {
+		appstore.InitBuiltin(builtin_hello.App())
+	}
+	err = appstore.Configure(conf)
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize the app store")
 	}
 
-	p.api = &api.Service{
-		Mattermost:   mm,
-		Configurator: conf,
-		Proxy:        proxy,
-		AppServices:  appservices.NewAppServices(mm, conf, store),
-		Admin:        admin.NewAdmin(mm, conf, store, proxy, awsClient, mutex),
-		AWS:          awsClient,
+	assetBucket := awsclient.GenerateS3BucketNameWithDefaults("")
```

Comment:
manifestBucket and assetBucket is the same.

Reply:
Yes, by design since they were currently the same, I didn't want to disturb it, just added potential de-coupling.